Refactor psa entropy inject test

orenc17 opened the pull request:
### Description
Replace TEST_ASSERT with more specific assertions
<!-- 
    Required
    Add here detailed changes summary, testing results, dependencies 
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->


### Pull request type

<!-- 
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [X] Test update
    [ ] Breaking change

### Reviewers

<!-- 
    Optional
    Request additional reviewers with @username
-->

ciarmcom commented:
@orenc17, thank you for your changes.
@ARMmbed/mbed-os-test @ARMmbed/mbed-os-crypto @ARMmbed/mbed-os-maintainers please review.

ciarmcom commented:
@orenc17, thank you for your changes.
@ARMmbed/mbed-os-test @ARMmbed/mbed-os-crypto @ARMmbed/mbed-os-maintainers please review.

Patater commented on TESTS/psa/entropy_inject/main.cpp:
```diff
@@ -90,6 +102,9 @@ utest::v1::status_t greentea_test_setup(const size_t number_of_cases)
     }
 
+    psa_status_t status = mbed_psa_reboot_and_request_new_security_state(PSA_LIFECYCLE_ASSEMBLY_AND_TEST);
+    TEST_ASSERT_EQUAL(PSA_SUCCESS, status);
```
This is a new test, not a refactor. Could you add it in a new commit?

Also, does this work? The function says "reboot", but if it actually reboots, what guarantee do we have that this test will resume after this function in the correct place?

orenc17 replied:
mbed_psa_reboot_and_request_new_security_state() API replaced its_reset() which is now a secure API only

This change is necessary for a clean environment for the test

Patater replied:
OK. We would benefit greatly from that being written down in the commit message that makes this change, a separate commit from the refactoring.

Does "reboot" actually reboot, like a soft-power reset?

Patater reviewed (commented):
Refactoring looks good, thanks.

I have questions about the new device lifetime function call and assert.

Commit:
Refactor psa entropy inject test

Replace TEST_ASSERT with more specific assertions

Commit:
Clear ITS environment in psa entropy tests

mbed_psa_reboot_and_request_new_security_state() API replaced its_reset() which is now a secure API only

This change is necessary for a clean environment for the test

orenc17 commented:
@Patater changed the commit message as requested

cmonr commented:
CI started

cmonr commented:
CI started.

~Will wait on @teetak01 to ack before merging~

Sorry, wrong PR.
